session/session.go

```diff
@@ -38,22 +28,27 @@ type Session struct {
 	Ticket string
 
 	hstop chan struct{}
+
+	// unavailableGuilds is a set of discord.Snowflakes of guilds that became
+	// unavailable when already connected to the gateway, i.e. sent in a
+	// GuildUnavailableEvent.
+	unavailableGuilds *moreatomic.SnowflakeSet
+	// unreadyGuilds is a set of discord.Snowflakes of guilds that were
+	// unavailable when connecting to the gateway, i.e. they had Unavailable
+	// set to true during Ready.
+	unreadyGuilds *moreatomic.SnowflakeSet
```

Comment:
Can this be moved to state? This does count as storing something, so session is a bad place to put it.